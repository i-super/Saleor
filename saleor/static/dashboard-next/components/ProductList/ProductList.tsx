import Cached from "material-ui-icons/Cached";
import Avatar from "material-ui/Avatar";
import { withStyles } from "material-ui/styles";
import Table, {
  TableBody,
  TableCell,
  TableFooter,
  TableHead,
  TableRow
} from "material-ui/Table";
import * as React from "react";

import { CategoryPropertiesQuery } from "../../gql-types";
import i18n from "../../i18n";
import Skeleton from "../Skeleton";
import TablePagination from "../TablePagination";

const decorate = withStyles(theme => ({
  avatarCell: {
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: 0,
    width: theme.spacing.unit * 5
  }
}));

interface ProductListProps {
  hasNextPage?: boolean;
  hasPreviousPage?: boolean;
  products?: Array<{
    id: string;
    name: string;
    productType: {
      name: string;
    };
    thumbnailUrl: string;
  }>;
  onNextPage();
  onPreviousPage();
}

export const ProductList = decorate<ProductListProps>(
  ({
    classes,
    hasNextPage,
    hasPreviousPage,
    onNextPage,
    onPreviousPage,
    products
  }) => (
    <Table>
      <TableHead>
        <TableRow>
          <TableCell className={classes.avatarCell} />
          <TableCell>{i18n.t("Name", { context: "object" })}</TableCell>
          <TableCell>{i18n.t("Type", { context: "object" })}</TableCell>
        </TableRow>
      </TableHead>
      <TableFooter>
        <TableRow>
          <TablePagination
            colSpan={3}
            hasNextPage={hasNextPage}
            onNextPage={onNextPage}
            hasPreviousPage={hasPreviousPage}
            onPreviousPage={onPreviousPage}
          />
        </TableRow>
      </TableFooter>
      <TableBody>
        {products === undefined || products === null ? (
          <TableRow>
            <TableCell className={classes.avatarCell}>
              <Avatar>
                <Cached />
              </Avatar>
            </TableCell>
            <TableCell>
              <Skeleton />
            </TableCell>
            <TableCell>
              <Skeleton />
            </TableCell>
          </TableRow>
        ) : products.length > 0 ? (
          products.map(product => (
            <TableRow key={product.id}>
              <TableCell className={classes.avatarCell}>
                <Avatar src={product.thumbnailUrl} />
              </TableCell>
              <TableCell>{product.name}</TableCell>
              <TableCell>{product.productType.name}</TableCell>
            </TableRow>
          ))
        ) : (
          <TableRow>
            <TableCell className={classes.avatarCell} />
            <TableCell colSpan={2}>{i18n.t("No products found")}</TableCell>
          </TableRow>
        )}
      </TableBody>
    </Table>
  )
);

export default ProductList;
