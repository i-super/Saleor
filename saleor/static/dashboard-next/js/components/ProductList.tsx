import Cached from "material-ui-icons/Cached";
import Avatar from "material-ui/Avatar";
import { withStyles } from "material-ui/styles";
import Table, {
  TableBody,
  TableCell,
  TableFooter,
  TableHead,
  TableRow
} from "material-ui/Table";
import * as React from "react";

import { CategoryPropertiesQuery } from "../gql-types";
import i18n from "../i18n";
import Skeleton from "./Skeleton";
import TablePagination from "./TablePagination";

const decorate = withStyles(theme => ({
  avatarCell: {
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: 0,
    width: theme.spacing.unit * 5
  }
}));

interface ProductListProps {
  hasNextPage?: boolean;
  hasPreviousPage?: boolean;
  products?: CategoryPropertiesQuery["category"]["products"]["edges"];
  onNextPage();
  onPreviousPage();
}

export const ProductList = decorate<ProductListProps>(
  ({
    classes,
    hasNextPage,
    hasPreviousPage,
    onNextPage,
    onPreviousPage,
    products
  }) => (
    <Table>
      <TableHead>
        <TableRow>
          <TableCell className={classes.avatarCell} />
          <TableCell>{i18n.t("Name", { context: "object" })}</TableCell>
          <TableCell>{i18n.t("Type", { context: "object" })}</TableCell>
        </TableRow>
      </TableHead>
      <TableFooter>
        <TableRow>
          <TablePagination
            colSpan={3}
            hasNextPage={hasNextPage}
            onNextPage={onNextPage}
            hasPreviousPage={hasPreviousPage}
            onPreviousPage={onPreviousPage}
          />
        </TableRow>
      </TableFooter>
      <TableBody>
        {products === undefined ? (
          <TableRow>
            <TableCell className={classes.avatarCell}>
              <Avatar>
                <Cached />
              </Avatar>
            </TableCell>
            <TableCell>
              <Skeleton />
            </TableCell>
            <TableCell>
              <Skeleton />
            </TableCell>
          </TableRow>
        ) : products.length > 0 ? (
          products.map(edge => (
            <TableRow key={edge.node.id}>
              <TableCell className={classes.avatarCell}>
                <Avatar src={edge.node.thumbnailUrl} />
              </TableCell>
              <TableCell>{edge.node.name}</TableCell>
              <TableCell>{edge.node.productType.name}</TableCell>
            </TableRow>
          ))
        ) : (
          <TableRow>
            <TableCell className={classes.avatarCell} />
            <TableCell colSpan={2}>{i18n.t("No products found")}</TableCell>
          </TableRow>
        )}
      </TableBody>
    </Table>
  )
);

export default ProductList;
