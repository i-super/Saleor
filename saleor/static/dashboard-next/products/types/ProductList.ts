/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { ProductFilterInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL query operation: ProductList
// ====================================================

export interface ProductList_products_edges_node_thumbnail {
  __typename: "Image";
  /**
   * The URL of the image.
   */
  url: string;
}

export interface ProductList_products_edges_node_basePrice {
  __typename: "Money";
  /**
   * Amount of money.
   */
  amount: number;
  /**
   * Currency code.
   */
  currency: string;
}

export interface ProductList_products_edges_node_productType {
  __typename: "ProductType";
  /**
   * The ID of the object.
   */
  id: string;
  name: string;
}

export interface ProductList_products_edges_node {
  __typename: "Product";
  /**
   * The ID of the object.
   */
  id: string;
  name: string;
  /**
   * The main thumbnail for a product.
   */
  thumbnail: ProductList_products_edges_node_thumbnail | null;
  /**
   * Whether the product is in stock and visible or not.
   */
  isAvailable: boolean | null;
  /**
   * The product's default base price.
   */
  basePrice: ProductList_products_edges_node_basePrice | null;
  productType: ProductList_products_edges_node_productType;
}

export interface ProductList_products_edges {
  __typename: "ProductCountableEdge";
  /**
   * The item at the end of the edge
   */
  node: ProductList_products_edges_node;
}

export interface ProductList_products_pageInfo {
  __typename: "PageInfo";
  /**
   * When paginating backwards, are there more items?
   */
  hasPreviousPage: boolean;
  /**
   * When paginating forwards, are there more items?
   */
  hasNextPage: boolean;
  /**
   * When paginating backwards, the cursor to continue.
   */
  startCursor: string | null;
  /**
   * When paginating forwards, the cursor to continue.
   */
  endCursor: string | null;
}

export interface ProductList_products {
  __typename: "ProductCountableConnection";
  edges: ProductList_products_edges[];
  /**
   * Pagination data for this connection.
   */
  pageInfo: ProductList_products_pageInfo;
}

export interface ProductList {
  /**
   * List of the shop's products.
   */
  products: ProductList_products | null;
}

export interface ProductListVariables {
  first?: number | null;
  after?: string | null;
  last?: number | null;
  before?: string | null;
  filter?: ProductFilterInput | null;
}
