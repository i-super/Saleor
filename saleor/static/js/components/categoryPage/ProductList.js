import React, { Component, PropTypes } from 'react';
import Relay from 'react-relay';

import ProductItem from './ProductItem';
import NoResults from './NoResults';

class ProductList extends Component {

  static propTypes = {
    onLoadMore: PropTypes.func.isRequired,
    products: PropTypes.object,
    setSorting: PropTypes.object
  };

  onLoadMore = () => this.props.onLoadMore();
  setSorting = (event) => this.props.setSorting(event);

  render() {
    const { edges, pageInfo: { hasNextPage } } = this.props.products;
    return (
      <div>
        <div className="row">
          {edges.length > 0 ? (edges.map((edge, i) => (
            <ProductItem key={i} product={edge.node} />
          ))) : (<NoResults />)}
        </div>
        <div className="load-more">
          {hasNextPage && (
            <button className="btn" onClick={this.onLoadMore}>Load more</button>
          )}
        </div>
      </div>
    );
  }
}

export default Relay.createContainer(ProductList, {
  fragments: {
    products: () => Relay.QL`
      fragment on ProductTypeConnection {
        edges {
          node {
            ${ProductItem.getFragment('product')}
          }
        }
        pageInfo {
          hasNextPage
        }
      }
    `
  }
});
