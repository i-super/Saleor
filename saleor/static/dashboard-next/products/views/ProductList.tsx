import Button from "@material-ui/core/Button";
import DialogContentText from "@material-ui/core/DialogContentText";
import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";
import * as React from "react";

import ActionDialog from "../../components/ActionDialog";
import { createPaginationState } from "../../components/Paginator";
import useBulkActions from "../../hooks/useBulkActions";
import useNavigator from "../../hooks/useNavigator";
import useNotifier from "../../hooks/useNotifier";
import usePaginator from "../../hooks/usePaginator";
import i18n from "../../i18n";
import { getMutationState, maybe } from "../../misc";
import { StockAvailability } from "../../types/globalTypes";
import ProductListCard from "../components/ProductListCard";
import { getTabName } from "../misc";
import {
  TypedProductBulkDeleteMutation,
  TypedProductBulkPublishMutation
} from "../mutations";
import { TypedProductListQuery } from "../queries";
import { productBulkDelete } from "../types/productBulkDelete";
import { productBulkPublish } from "../types/productBulkPublish";
import {
  productAddUrl,
  productListUrl,
  ProductListUrlDialog,
  ProductListUrlFilters,
  ProductListUrlQueryParams,
  productUrl
} from "../urls";

interface ProductListProps {
  params: ProductListUrlQueryParams;
}

const PAGINATE_BY = 20;

export const ProductList: React.StatelessComponent<ProductListProps> = ({
  params
}) => {
  const navigate = useNavigator();
  const notify = useNotifier();
  const paginate = usePaginator();
  const { isSelected, listElements, reset, toggle } = useBulkActions(
    params.ids
  );

  const closeModal = () =>
    navigate(
      productListUrl({
        ...params,
        action: undefined,
        ids: undefined
      }),
      true
    );

  const changeFilters = (filters: ProductListUrlFilters) => {
    reset();
    navigate(productListUrl(filters));
  };

  const openModal = (action: ProductListUrlDialog, ids: string[]) =>
    navigate(
      productListUrl({
        ...params,
        action,
        ids
      })
    );

  const paginationState = createPaginationState(PAGINATE_BY, params);

  return (
    <TypedProductListQuery
      displayLoader
      variables={{
        ...paginationState,
        stockAvailability: params.status
      }}
    >
      {({ data, loading, refetch }) => {
        const currentTab = getTabName(params);
        const { loadNextPage, loadPreviousPage, pageInfo } = paginate(
          maybe(() => data.products.pageInfo),
          paginationState,
          params
        );

        const handleBulkDelete = (data: productBulkDelete) => {
          if (data.productBulkDelete.errors.length === 0) {
            closeModal();
            notify({
              text: i18n.t("Products removed")
            });
            reset();
            refetch();
          }
        };

        const handleBulkPublish = (data: productBulkPublish) => {
          if (data.productBulkPublish.errors.length === 0) {
            closeModal();
            notify({
              text: i18n.t("Changed publication status")
            });
            reset();
            refetch();
          }
        };

        return (
          <TypedProductBulkDeleteMutation onCompleted={handleBulkDelete}>
            {(productBulkDelete, productBulkDeleteOpts) => (
              <TypedProductBulkPublishMutation onCompleted={handleBulkPublish}>
                {(productBulkPublish, productBulkPublishOpts) => {
                  const bulkDeleteMutationState = getMutationState(
                    productBulkDeleteOpts.called,
                    productBulkDeleteOpts.loading,
                    maybe(
                      () => productBulkDeleteOpts.data.productBulkDelete.errors
                    )
                  );

                  const bulkPublishMutationState = getMutationState(
                    productBulkPublishOpts.called,
                    productBulkPublishOpts.loading,
                    maybe(
                      () =>
                        productBulkPublishOpts.data.productBulkPublish.errors
                    )
                  );

                  return (
                    <>
                      <ProductListCard
                        currentTab={currentTab}
                        filtersList={[]}
                        onAdd={() => navigate(productAddUrl)}
                        disabled={loading}
                        products={
                          data &&
                          data.products !== undefined &&
                          data.products !== null
                            ? data.products.edges.map(p => p.node)
                            : undefined
                        }
                        onNextPage={loadNextPage}
                        onPreviousPage={loadPreviousPage}
                        pageInfo={pageInfo}
                        onRowClick={id => () => navigate(productUrl(id))}
                        onAllProducts={() =>
                          changeFilters({
                            status: undefined
                          })
                        }
                        onCustomFilter={() => undefined}
                        onAvailable={() =>
                          changeFilters({
                            status: StockAvailability.IN_STOCK
                          })
                        }
                        onOfStock={() =>
                          changeFilters({
                            status: StockAvailability.OUT_OF_STOCK
                          })
                        }
                        toolbar={
                          <>
                            <Button
                              color="primary"
                              onClick={() =>
                                openModal("unpublish", listElements)
                              }
                            >
                              {i18n.t("Unpublish")}
                            </Button>
                            <Button
                              color="primary"
                              onClick={() => openModal("publish", listElements)}
                            >
                              {i18n.t("Publish")}
                            </Button>
                            <IconButton
                              color="primary"
                              onClick={() => openModal("delete", listElements)}
                            >
                              <DeleteIcon />
                            </IconButton>
                          </>
                        }
                        isChecked={isSelected}
                        selected={listElements.length}
                        toggle={toggle}
                      />
                      <ActionDialog
                        open={params.action === "delete"}
                        confirmButtonState={bulkDeleteMutationState}
                        onClose={closeModal}
                        onConfirm={() =>
                          productBulkDelete({ variables: { ids: params.ids } })
                        }
                        title={i18n.t("Remove products")}
                        variant="delete"
                      >
                        <DialogContentText
                          dangerouslySetInnerHTML={{
                            __html: i18n.t(
                              "Are you sure you want to remove <strong>{{ number }}</strong> products?",
                              {
                                number: maybe(
                                  () => params.ids.length.toString(),
                                  "..."
                                )
                              }
                            )
                          }}
                        />
                      </ActionDialog>
                      <ActionDialog
                        open={params.action === "publish"}
                        confirmButtonState={bulkPublishMutationState}
                        onClose={closeModal}
                        onConfirm={() =>
                          productBulkPublish({
                            variables: {
                              ids: params.ids,
                              isPublished: true
                            }
                          })
                        }
                        title={i18n.t("Publish products")}
                      >
                        <DialogContentText
                          dangerouslySetInnerHTML={{
                            __html: i18n.t(
                              "Are you sure you want to publish <strong>{{ number }}</strong> products?",
                              {
                                number: maybe(
                                  () => params.ids.length.toString(),
                                  "..."
                                )
                              }
                            )
                          }}
                        />
                      </ActionDialog>
                      <ActionDialog
                        open={params.action === "unpublish"}
                        confirmButtonState={bulkPublishMutationState}
                        onClose={closeModal}
                        onConfirm={() =>
                          productBulkPublish({
                            variables: {
                              ids: params.ids,
                              isPublished: false
                            }
                          })
                        }
                        title={i18n.t("Unpublish products")}
                      >
                        <DialogContentText
                          dangerouslySetInnerHTML={{
                            __html: i18n.t(
                              "Are you sure you want to unpublish <strong>{{ number }}</strong> products?",
                              {
                                number: maybe(
                                  () => params.ids.length.toString(),
                                  "..."
                                )
                              }
                            )
                          }}
                        />
                      </ActionDialog>
                    </>
                  );
                }}
              </TypedProductBulkPublishMutation>
            )}
          </TypedProductBulkDeleteMutation>
        );
      }}
    </TypedProductListQuery>
  );
};
export default ProductList;
