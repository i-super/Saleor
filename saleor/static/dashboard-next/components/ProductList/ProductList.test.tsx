import * as React from "react";
import * as renderer from "react-test-renderer";

import ProductList from "./ProductList";

const products = [
  {
    id: "prod1",
    name: "Lewis PLC",
    productType: {
      id: "123412341234",
      name: "T-sirts"
    },
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/5_ZoF6Xi8-crop-c0-5__0-5-255x255-70.jpg"
  },
  {
    id: "prod2",
    name: "Kennedy-Ramirez",
    productType: {
      id: "123412341234",
      name: "T-sirts"
    },
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_07Fa6v8-crop-c0-5__0-5-255x255-70.jpg"
  },
  {
    id: "prod3",
    name: "Newman, Ashley and Roberson",
    productType: {
      id: "123412341234",
      name: "T-sirts"
    },
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_zeczDly-crop-c0-5__0-5-255x255-70.jpg"
  }
];

describe("<ProductList />", () => {
  it("renders while data is loading", () => {
    const component = renderer.create(
      <ProductList
        hasPreviousPage={false}
        hasNextPage={false}
        onNextPage={jest.fn()}
        onPreviousPage={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders properly when data is loaded", () => {
    const component = renderer.create(
      <ProductList
        hasPreviousPage={false}
        hasNextPage={false}
        onNextPage={jest.fn()}
        onPreviousPage={jest.fn()}
        products={products}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders properly when product list is empty", () => {
    const component = renderer.create(
      <ProductList
        hasNextPage={false}
        onNextPage={jest.fn()}
        onPreviousPage={jest.fn()}
        products={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
