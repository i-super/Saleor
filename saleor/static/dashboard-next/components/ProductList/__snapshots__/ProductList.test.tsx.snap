// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> renders properly when data is loaded 1`] = `
<table
  className="MuiTable-root-2"
>
  <thead
    className="MuiTableHead-root-3"
  >
    <tr
      className="MuiTableRow-root-4 MuiTableRow-head-5"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10 Component-avatarCell-1"
        scope="col"
      />
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10"
        scope="col"
      >
        Type
      </th>
    </tr>
  </thead>
  <tfoot>
    <tr
      className="MuiTableRow-root-4 MuiTableRow-footer-6"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-footer-12 MuiTablePagination-root-20"
        colSpan={3}
        scope={undefined}
      >
        <div
          className="MuiToolbar-root-26 MuiToolbar-gutters-27 MuiTablePagination-toolbar-25"
        >
          <div
            className="MuiTablePagination-spacer-24"
          />
          <div
            className="MuiTablePaginationActions-root-28"
          >
            <button
              className="MuiButtonBase-root-35 MuiButtonBase-disabled-36 MuiIconButton-root-29 MuiIconButton-disabled-33"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-34"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-38"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              className="MuiButtonBase-root-35 MuiButtonBase-disabled-36 MuiIconButton-root-29 MuiIconButton-disabled-33"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-34"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-38"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
  <tbody
    className="MuiTableBody-root-44"
  >
    <tr
      className="MuiTableRow-root-4"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11 Component-avatarCell-1"
        scope={undefined}
      >
        <div
          className="MuiAvatar-root-45"
        >
          <img
            alt={undefined}
            className="MuiAvatar-img-47"
            sizes={undefined}
            src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_ZoF6Xi8-crop-c0-5__0-5-255x255-70.jpg"
            srcSet={undefined}
          />
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        Lewis PLC
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        T-sirts
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-4"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11 Component-avatarCell-1"
        scope={undefined}
      >
        <div
          className="MuiAvatar-root-45"
        >
          <img
            alt={undefined}
            className="MuiAvatar-img-47"
            sizes={undefined}
            src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_07Fa6v8-crop-c0-5__0-5-255x255-70.jpg"
            srcSet={undefined}
          />
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        Kennedy-Ramirez
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        T-sirts
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-4"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11 Component-avatarCell-1"
        scope={undefined}
      >
        <div
          className="MuiAvatar-root-45"
        >
          <img
            alt={undefined}
            className="MuiAvatar-img-47"
            sizes={undefined}
            src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_zeczDly-crop-c0-5__0-5-255x255-70.jpg"
            srcSet={undefined}
          />
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        Newman, Ashley and Roberson
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        T-sirts
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<ProductList /> renders properly when product list is empty 1`] = `
<table
  className="MuiTable-root-2"
>
  <thead
    className="MuiTableHead-root-3"
  >
    <tr
      className="MuiTableRow-root-4 MuiTableRow-head-5"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10 Component-avatarCell-1"
        scope="col"
      />
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10"
        scope="col"
      >
        Type
      </th>
    </tr>
  </thead>
  <tfoot>
    <tr
      className="MuiTableRow-root-4 MuiTableRow-footer-6"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-footer-12 MuiTablePagination-root-20"
        colSpan={3}
        scope={undefined}
      >
        <div
          className="MuiToolbar-root-26 MuiToolbar-gutters-27 MuiTablePagination-toolbar-25"
        >
          <div
            className="MuiTablePagination-spacer-24"
          />
          <div
            className="MuiTablePaginationActions-root-28"
          >
            <button
              className="MuiButtonBase-root-35 MuiButtonBase-disabled-36 MuiIconButton-root-29 MuiIconButton-disabled-33"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-34"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-38"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              className="MuiButtonBase-root-35 MuiButtonBase-disabled-36 MuiIconButton-root-29 MuiIconButton-disabled-33"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-34"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-38"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
  <tbody
    className="MuiTableBody-root-44"
  >
    <tr
      className="MuiTableRow-root-4"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11 Component-avatarCell-1"
        scope={undefined}
      />
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        colSpan={2}
        scope={undefined}
      >
        No products found
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<ProductList /> renders while data is loading 1`] = `
<table
  className="MuiTable-root-2"
>
  <thead
    className="MuiTableHead-root-3"
  >
    <tr
      className="MuiTableRow-root-4 MuiTableRow-head-5"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10 Component-avatarCell-1"
        scope="col"
      />
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-head-10"
        scope="col"
      >
        Type
      </th>
    </tr>
  </thead>
  <tfoot>
    <tr
      className="MuiTableRow-root-4 MuiTableRow-footer-6"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-footer-12 MuiTablePagination-root-20"
        colSpan={3}
        scope={undefined}
      >
        <div
          className="MuiToolbar-root-26 MuiToolbar-gutters-27 MuiTablePagination-toolbar-25"
        >
          <div
            className="MuiTablePagination-spacer-24"
          />
          <div
            className="MuiTablePaginationActions-root-28"
          >
            <button
              className="MuiButtonBase-root-35 MuiButtonBase-disabled-36 MuiIconButton-root-29 MuiIconButton-disabled-33"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-34"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-38"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              className="MuiButtonBase-root-35 MuiButtonBase-disabled-36 MuiIconButton-root-29 MuiIconButton-disabled-33"
              disabled={true}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiIconButton-label-34"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-38"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
  <tbody
    className="MuiTableBody-root-44"
  >
    <tr
      className="MuiTableRow-root-4"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11 Component-avatarCell-1"
        scope={undefined}
      >
        <div
          className="MuiAvatar-root-45 MuiAvatar-colorDefault-46"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-38"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 8l-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3l-4-4zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4H6z"
            />
          </svg>
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        <span
          className="Component-skeleton-48"
          style={undefined}
        >
          ‌
        </span>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-9 MuiTableCell-body-11"
        scope={undefined}
      >
        <span
          className="Component-skeleton-48"
          style={undefined}
        >
          ‌
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
